### PR TITLE
mgr/restful: updated string formatting to str.format()

### PR DESCRIPTION
**Pull request by falcon78921**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

Updated string formatting to Python3-style ``str.format()``. Removed older string formatting (``%`` operator).

Signed-off-by: James McClune <jmcclune@mcclunetechnologies.net>

### CODE REVIEW

**Comment by falcon78921** [comment]
jenkins retest this please

**Comment by alfredodeza** [comment]
I don't think this makes it "python3" as it is valid. Could you expand on the reasoning?

**Comment by falcon78921** [comment]
@alfredodeza I agree that it's not so much "Python3". I meant with the newer releases of Python, ``str.format`` is  becoming a more pervasive and cleaner approach to string formatting.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-02-09_03:55:51-rados-wip-kefu2-testing-2019-02-06-2219-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-02-07_05:29:45-rados-wip-kefu2-testing-2019-02-06-2219-distro-basic-mira/

the failure is tracked by http://tracker.ceph.com/issues/38249

File: src/pybind/mgr/restful/api/config.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def patch(self, **kwargs):
         valid_flags = set(args.keys()) & set(common.OSD_FLAGS)
         invalid_flags = list(set(args.keys()) - valid_flags)
         if invalid_flags:
-            context.instance.log.warn("%s not valid to set/unset" % invalid_flags)
+            context.instance.log.warn("{} not valid to set/unset".format(invalid_flags))
```
Comment (tchaikov) [inline src/pybind/mgr/restful/api/config.py]: probably a more pythonic change is to put
```py
context.instance.log.warn("%s not valid to set/unset", invalid_flags)
```

**Comment by tchaikov** [comment]
> I meant with the newer releases of Python, ``str.format()`` is  becoming a more pervasive and cleaner approach to string formatting.

i agree that the `str.format()` method is relatively new than `%`. but i think it'd be better if we can accompany these "modernizing" changes with some other more substantial changes like refactor, bug fixing or new features.

File: src/pybind/mgr/restful/api/config.py (hunk 1 above)
Comment (falcon78921) [reply src/pybind/mgr/restful/api/config.py]: @tchaikov I can make that change. As a test, I created a list (like the ``OSD_FLAGS`` list in ``common.py``):

``flags = ['osdflag1']``

If I modify the code like you recommended, would I get this as a return, being that someone threw an ``OSD_FLAG`` that was not valid?

``%s not valid to set/unset ['osdflag1']``

Comment (tchaikov) [reply src/pybind/mgr/restful/api/config.py]: @falcon78921 please see https://docs.python.org/3/library/logging.html#logging.debug for the answer.

Comment (falcon78921) [reply src/pybind/mgr/restful/api/config.py]: @tchaikov Thank you for the explanation Kefu. I made the change.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-02-14_06:27:37-rados-wip-kefu2-testing-2019-02-14-1156-distro-basic-smithi/

**Pull request merged**